### ISSUE TITLE
void

### ISSUE THREAD

**Issue by abebeos** [bitfocus/companion#1932]

**Comment by krocheck** [comment]
LOL ... we go back and forth on that.  There were/are quite major changes in 2.2 compared to 2.1 (including a new ground up UI in React).  Think of 2.3 as a 'code name' for now.

You might want to join our [Slack](https://join.slack.com/t/bitfocusio/shared_invite/zt-o1i0mohx-p_m5I7fUEUJ1BBu_KrDN2Q).  A lot of active discussion happens over there.

**Comment by Julusian** [comment]
Im often referring to it as 2.3 to indicate that this will be the next release after 2.2, as 3.0 is vaguer as to when that will be.

**Comment by abebeos** [comment]
> You might want to join our [Slack](https://join.slack.com/t/bitfocusio/shared_invite/zt-o1i0mohx-p_m5I7fUEUJ1BBu_KrDN2Q). A lot of active discussion happens over there.

Thanks for the invite. For now I need to stick to issues, to avoid an "information-overload". Chat bandwidth sometimes knocks me down, especially when I'm new to a system.

> 2.3 to indicate that this will be the next release after 2.2

I see. As things look to me (99,5% confidence), you should go with `3.0`.

Further benefit: you're then free to have 2.2, 2.3, 2.4, ..., 2.9 versions etc., for intermediate minor api-breaking steps towards 3.0.

**Comment by krocheck** [comment]
> Further benefit: you're then free to have 2.2, 2.3, 2.4, ..., 2.9 versions etc., for intermediate minor api-breaking steps towards 3.0.

Given my history with merge conflicts becoming way too difficult such that you abandon work, I don't have interest in there being 2.x and 3.x development in parallel.  We've setup new processes so that we can do this core dev work off to the side while module work and core bug fixes can happen in parallel, but anything beyond that I think will be counterproductive.

**Comment by abebeos** [comment]
> I don't have interest in there being 2.x and 3.x development in parallel.

Parallel development is not about "having interest in", but about "what reality dictates".

As things stand, I don't see how one could avoid parallel development (but this is really another issue). 

Main issue here:

>> I see. As things look to me (99,5% confidence), you should go with `3.0`.

This issue here is really only about the `3.0` labeling/versioning (use `3.0` instead of `2.3` when referring to the redesigned companion)

**Comment by Julusian** [comment]
> Parallel development is not about "having interest in", but about "what reality dictates".
> 
> As things stand, I don't see how one could avoid parallel development (but this is really another issue).

As I see it, it only matters if the refactor takes ends up taking too long and there is some feature that we really want to get to users that we dont think we can squeeze into a 2.2.x release.

But in the unlikely scenario that something does make us want to do a release before 3.x then we will need to think about our git branches, and how do we label beta/testing builds. Its not just a matter of the version number

**Comment by abebeos** [comment]
sounds ok.

**Comment by abebeos** [comment]
> But in the unlikely scenario that something does make us want to do a release before 3.x

If you (core-devs) agree on the `3.0` label for the redesigned companion, then it would be nice to see a milestone (or milestones) to see the issues assigned to. Personally I depend quite much on having a 'roadmap', build by a few open milestones. Or at least those github-projects.

* Milestone 2.1.x (maintenance release)
  * Issue A
  * Issue B 
  
* Milestone 3.0.0 (Companion Redesign)

(would be another issue to file, but i don't want to 'bombard' you...)

**Comment by Julusian** [comment]
There is a Project containing some things I would like to personally do for 3.0 https://github.com/orgs/bitfocus/projects/2 but I probably won't get through it all. Essentially that is my todo list currently, but its hard to make a real roadmap because it also falls down to what I am motivated to work on. I work on this in my free-time, so I mostly keep to things I am interested in doing

As for maintenance releases, I generally try to fix bugs as they are opened (if they interest me) or they will be fixed by someone at some point in the future. So a milestone for a maintenance release will typically have nothing in it.

**Comment by abebeos** [comment]
> I work on this in my free-time, so I mostly keep to things I am interested in doing

Ok, wasn't aware of this.

> falls down to what I am motivated to work on.

I understand completely, motivating oneself is possibly the most difficult thing in development.

-

There are currently 246 open issues, possibly a triage via a version-label should be enough. It is otherwise difficult to get an overview.

**Comment by krocheck** [comment]
> I work on this in my free-time, so I mostly keep to things I am interested in doing

Ditto.

> There are currently 246 open issues, possibly a triage via a version-label should be enough. It is otherwise difficult to get an overview.

We get to them when we get to them.  This is an open source project with a volunteer dev team.

**Comment by abebeos** [comment]
> We get to them when we get to them. This is an open source project with a volunteer dev team.

I see.

> There is a Project containing some things I would like to personally do for 3.0 https://github.com/orgs/bitfocus/projects/2 but I probably won't get through it all. Essentially that is my todo list currently,

If you find the time to file issues for your ongoing work, and assign them to your "todo" project, that would help me already very much.
